java/src/main/java/com/logicalclocks/hsfs/engine/TrainingDatasetEngine.java

```diff
@@ -294,4 +306,19 @@ private Object deserializeComplexFeature(Map<String, DatumReader<Object>> comple
   public void delete(TrainingDataset trainingDataset) throws FeatureStoreException, IOException {
     trainingDatasetApi.delete(trainingDataset);
   }
+
+  private List<TrainingDatasetFeature> getTransformationFunction(TrainingDataset trainingDataset)
```

Comment:
In python it's called (more appropriately IMHO) - `getTransformationFunctions` - you return a list here, it should be plural.